bindings/python/tests/test_parse.py

```diff
@@ -0,0 +1,332 @@
+    assert cxx_base_specifier["name"] == "struct BaseStruct"
+
+
+def test_cxx_method(tmp_path):
+    file_contents = """
+    struct AStruct {
+        void aMethod() {}
+    };
+    """
+    parsed_info = get_parsed_info(tmp_path=tmp_path, file_contents=file_contents)
+
+    struct_decl = parsed_info["members"][0]
+    cxx_method = struct_decl["members"][0]
+
+    assert cxx_method["kind"] == "CXX_METHOD"
+    assert cxx_method["result_type"] == "void"
+    assert cxx_method["name"] == "aMethod"
+
+
+def test_var_decl(tmp_path):
+    file_contents = "int anInt = 1;"
+    parsed_info = get_parsed_info(tmp_path=tmp_path, file_contents=file_contents)
+
+    var_decl = parsed_info["members"][0]
+
+    assert var_decl["kind"] == "VAR_DECL"
+    assert var_decl["element_type"] == "Int"
+    assert var_decl["name"] == "anInt"
+
+
+def test_type_ref(tmp_path):
+    file_contents = """
+    struct AStruct { 
+        int aClassMember;
+        void aMethod(AStruct& aClassMember) {}
+    };
+    """
+    parsed_info = get_parsed_info(tmp_path=tmp_path, file_contents=file_contents)
+
+    struct_decl = parsed_info["members"][0]
+    cxx_method = struct_decl["members"][1]
+    l_value_ref = cxx_method["members"][0]
+    type_ref = l_value_ref["members"][0]
+
+    assert type_ref["kind"] == "TYPE_REF"
+    assert type_ref["name"] == "struct AStruct"
+
+
+def test_constructor(tmp_path):
+    file_contents = """
+    struct AStruct {
+        AStruct() {}
+    };
+    """
+    parsed_info = get_parsed_info(tmp_path=tmp_path, file_contents=file_contents)
+
+    struct_decl = parsed_info["members"][0]
+    constructor = struct_decl["members"][0]
+
+    assert constructor["kind"] == "CONSTRUCTOR"
+    assert constructor["access_specifier"] == "PUBLIC"
+    assert constructor["name"] == "AStruct"
+
+
+def test_parm_decl(tmp_path):
+    file_contents = """
+    struct AStruct {
+        AStruct(int aFunctionArgument) {}
+    };
+    """
+    parsed_info = get_parsed_info(tmp_path=tmp_path, file_contents=file_contents)
+
+    struct_decl = parsed_info["members"][0]
+    constructor = struct_decl["members"][0]
+    parm_decl = constructor["members"][0]
+
+    assert parm_decl["kind"] == "PARM_DECL"
+    assert parm_decl["element_type"] == "Int"
+    assert parm_decl["name"] == "aFunctionArgument"
+
+
+def test_call_expr(tmp_path):
+    file_contents = """
+    struct AStruct {
+        AStruct() : AStruct() {}
+    };
+    """
+    parsed_info = get_parsed_info(tmp_path=tmp_path, file_contents=file_contents)
+
+    struct_decl = parsed_info["members"][0]
+    constructor = struct_decl["members"][0]
+    call_expr = constructor["members"][1]
+
+    assert call_expr["kind"] == "CALL_EXPR"
+    assert call_expr["name"] == "AStruct"
+
+
+def test_unexposed_expr(tmp_path):
+    file_contents = """
+    struct AStruct {
+        int aClassMember;
+        AStruct(int aFunctionArgument) : aClassMember(aFunctionArgument) {}
```

Comment:
This example raises questions regarding:
```py
AStruct(int firstArg, int secondArg): firstMember(secondArg), secondMember(firstArg) {}
```

Reply:
It works.
Add this one too?

Reply:
The question is would `secondMember` be the 3rd element or the 2nd (aka index 2 or 1)?

Reply:
```diff
- ...
                        {
                            "depth": 3,
                            "line": 4,
                            "column": 86,
                            "kind": "UNEXPOSED_EXPR",
                            "name": "secondFunctionArgument",
                            "element_type": "Int",
                            "members": [
                                {
                                    "depth": 4,
                                    "line": 4,
                                    "column": 86,
                                    "kind": "DECL_REF_EXPR",
                                    "name": "secondFunctionArgument",
                                    "element_type": "Int",
                                    "members": [],
                                }
                            ],
                        },
- ...
                        {
                            "depth": 3,
                            "line": 4,
                            "column": 124,
                            "kind": "UNEXPOSED_EXPR",
                            "name": "firstFunctionArgument",
                            "element_type": "Int",
                            "members": [
                                {
                                    "depth": 4,
                                    "line": 4,
                                    "column": 124,
                                    "kind": "DECL_REF_EXPR",
                                    "name": "firstFunctionArgument",
                                    "element_type": "Int",
                                    "members": [],
                                }
- ...
```

This should be tested at decl_ref_expr, instead of unexposed_expr
Something like this:

```py
def test_decl_ref_expr(tmp_path):
    file_contents = """
    struct AStruct { 
        int firstMember, secondMember;
        AStruct(int firstFunctionArgument, int secondFunctionArgument) : firstMember(secondFunctionArgument), secondMember(firstFunctionArgument) {}
    };
    """
    parsed_info = get_parsed_info(tmp_path=tmp_path, file_contents=file_contents)

    struct_decl = parsed_info["members"][0]
    constructor = struct_decl["members"][2]
    unexposed_expr_1 = constructor["members"][3]
    unexposed_expr_2 = constructor["members"][5]
    decl_ref_expr_1 = unexposed_expr_1["members"][0]
    decl_ref_expr_2 = unexposed_expr_2["members"][0]

    assert decl_ref_expr_1["kind"] == "DECL_REF_EXPR"
    assert decl_ref_expr_2["kind"] == "DECL_REF_EXPR"
    assert decl_ref_expr_1["name"] == "secondFunctionArgument"
    assert decl_ref_expr_2["name"] == "firstFunctionArgument"
```

For member ref:
```py
def test_member_ref(tmp_path):
    file_contents = """
    struct AStruct { 
        int firstMember, secondMember;
        AStruct(int firstFunctionArgument, int secondFunctionArgument) : firstMember(secondFunctionArgument), secondMember(firstFunctionArgument) {}
    };
    """
    parsed_info = get_parsed_info(tmp_path=tmp_path, file_contents=file_contents)
    struct_decl = parsed_info["members"][0]
    constructor = struct_decl["members"][2]
    member_ref_1 = constructor["members"][2]
    member_ref_2 = constructor["members"][4]

    assert member_ref_1["kind"] == "MEMBER_REF"
    assert member_ref_2["kind"] == "MEMBER_REF"
    assert member_ref_1["element_type"] == "Int"
    assert member_ref_2["element_type"] == "Int"
    assert member_ref_1["name"] == "firstMember"
    assert member_ref_2["name"] == "secondMember"
```